rtl8720dn/wioterminal.go

```diff
@@ -0,0 +1,67 @@
+//go:build wioterminal
+// +build wioterminal
+
+package rtl8720dn
+
+import (
+	"device/sam"
+	"machine"
+	"runtime/interrupt"
+	"time"
+)
+
+func NewForWioterminal() *Driver {
+	initWioterminal()
+
+	return &Driver{
+		RTL8720DN: &RTL8720DN {
+			port:  UART_RTL8720DN,
+			seq:   1,
+			sema:  make(chan bool, 1),
+			debug: false,
+		},
+	}
+}
+
+// UART_RTL8720DN is the WioTerminal's SPI port that is connected to the 
+// RTL8720DN device. To use it is required to reconfigure the SPI as a UART.
+var (
+	UART_RTL8720DN *UARTx
+)
+
+func handleInterrupt(interrupt.Interrupt) {
+	// should reset IRQ
+	UART_RTL8720DN.Receive(byte((UART_RTL8720DN.Bus.DATA.Get() & 0xFF)))
+	UART_RTL8720DN.Bus.INTFLAG.SetBits(sam.SERCOM_USART_INT_INTFLAG_RXC)
+}
+
+func initWioterminal() {
+	machine.RTL8720D_CHIP_PU.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	machine.RTL8720D_CHIP_PU.Low()
+	time.Sleep(100 * time.Millisecond)
+	machine.RTL8720D_CHIP_PU.High()
+	time.Sleep(1000 * time.Millisecond)
+
+	UART_RTL8720DN = &UARTx{
+		UART: &machine.UART{
+			Buffer: machine.NewRingBuffer(),
+			Bus:    sam.SERCOM0_USART_INT,
+			SERCOM: 0,
+		},
+	}
```

Comment:
This part is also required for non-WioTerminal.
The important parts are the following.

```
func (u *UARTx) Read(p []byte) (n int, err error) {
	if u.Buffered() == 0 {
		time.Sleep(1 * time.Millisecond)
		return 0, nil
	}
	return u.UART.Read(p)
}
```

The UARTx definition can be moved to `rtl8720dn/rtl8720dn.go` since it is a common process even at non-Wio Terminal.

Reply:
Done.